packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx

```diff
@@ -107,14 +107,11 @@ const OrderDetailsTable: React.FC<OrderDetailsProps> = ({ title, patientUuid, sh
         case 'drugorder':
           launchAddDrugOrder();
           break;
-        case 'testorder':
-          launchAddLabsOrder();
```

Comment:
Couldn't this have been solved by changing this to:

```ts
launchAddLabsOrder({order: orderItem});
```

That seems like a cleaner immplementation to me. Closing the workspace seems like it would potentially have undesirable side-effects.

Reply:
Hi @ibacher I have removed the implementation to close the workspace and made another push. Thanks

Reply:
I'd still like if we could go back to using this switch. It makes it easier when we need to implement additional order types.

Reply:
Hi @ibacher reverted to the switch statement; implementation is slightly different from the suggested `launchAddLabsOrder({order: orderItem});` cause it was loading a blank form. Thanks